src/yz_solrq_worker.erl

```diff
@@ -171,89 +167,91 @@ cancel_drain(QPid) ->
--spec request_batch(solrq_id(), index_name(), solrq_helper_id()) -> ok.
-request_batch(QPid, Index, HPid) ->
-    gen_server:cast(QPid, {request_batch, Index, HPid}).
+-spec request_batch(solrq_id(), solrq_helper_id()) -> ok.
+request_batch(QPid, HPid) ->
+    gen_server:cast(QPid, {request_batch, HPid}).
 
 
 -spec batch_complete(
     solrq_id(),
-    index_name(),
     {NumDelivered :: non_neg_integer(), ok} |
     {NumDelivered :: non_neg_integer(),
      {retry, [Undelivered :: solrq_message()]}}) -> ok.
-batch_complete(QPid, Index, Message) ->
-    gen_server:cast(QPid, {batch_complete, Index, Message}).
+batch_complete(QPid, Message) ->
+    gen_server:cast(QPid, {batch_complete, Message}).
 
 %%%===================================================================
 %%% gen_server callbacks
 %%%===================================================================
 
 %% @private
-init([]) ->
-    {ok, read_appenv(#state{})} .
-
-handle_call({index, Index, E}, From, State) ->
-    ?PULSE_DEBUG("index.  State: ~p~n", [debug_state(State)]),
-    State2 = inc_qlen_and_maybe_unblock_vnode(From, State),
-    IndexQ = enqueue(E, get_indexq(Index, State2)),
-    IndexQ2 = maybe_request_worker(Index, IndexQ),
-    IndexQ3 = maybe_start_timer(Index, IndexQ2),
-    NewState = update_indexq(Index, IndexQ3, State2),
-    ?PULSE_DEBUG("index.  NewState: ~p~n", [debug_state(NewState)]),
-    {noreply, NewState};
+init([Index, Partition]) ->
+    State0 = read_appenv(#state{index = Index, partition = Partition}),
+    State1 = get_helper(Index, Partition, State0),
+    {ok, State1}.
+
+handle_call({index, E}, From, State0) ->
+    ?PULSE_DEBUG("index.  State: ~p~n", [debug_state(State0)]),
+    State1 = maybe_purge(State0),
+    State2 = maybe_send_reply(From, State1),
+    State3 = enqueue(E, State2),
+    State4 = maybe_send_batch_to_helper(State3),
+    FinalState = maybe_start_timer(State4),
+    ?PULSE_DEBUG("index.  NewState: ~p~n", [debug_state(FinalState)]),
+    {noreply, FinalState};
 handle_call(status, _From, #state{} = State) ->
     {reply, internal_status(State), State};
 handle_call({set_hwm, NewHWM}, _From, #state{queue_hwm = OldHWM} = State) ->
     {reply, {ok, OldHWM}, maybe_unblock_vnodes(State#state{queue_hwm = NewHWM})};
 handle_call(get_hwm, _From, #state{queue_hwm = HWM} = State) ->
     {reply, HWM, State};
-handle_call({set_index, Index, Min, Max, DelayMS}, _From, State) ->
-    IndexQ = get_indexq(Index, State),
-    IndexQ2 = maybe_request_worker(Index,
-                                   IndexQ#indexq{batch_min = Min,
-                                                 batch_max = Max,
-                                                 delayms_max = DelayMS}),
-    OldParams = {IndexQ#indexq.batch_min,
-                 IndexQ#indexq.batch_max,
-                 IndexQ#indexq.delayms_max},
-    {reply, {ok, OldParams}, update_indexq(Index, IndexQ2, State)};
+handle_call({set_index, Min, Max, DelayMS}, _From,
+            State0) ->
```

Comment:
Probably could just put the above function head on one line instead of splitting it. Also missing a space before the equals sign on the next line below.